File: code/python/removeDuplicatesII.py
class Solution:
    def removeDuplicatesII(self, nums: List[int]) -> int:
    '''The function name was changed to avoid duplicate in file names'''
        index = 1
        length = len(nums)
        if length <= 1:
            return length
        
        while index < length-1:
            if nums[index] == nums[index-1] and nums[index] == nums[index+1]:
                nums.pop(index)
                length -= 1
            else:
                index += 1
        return length

File: code/java/removeDuplicatesII.java
class Solution {
    public int removeDuplicatesII(int[] nums) {
        if(nums.length < 3)
            return nums.length;
        
        int temp = nums[0] - 1; 
        
        for(int i=0;i<nums.length-2;i++){
            if(nums[i+2] == nums[i+1] && nums[i] == nums[i+1])
                nums[i] = temp;
        }
        
        int j = 0;
        for(int i=0;i<nums.length;i++)
            if(nums[i] != temp){
                nums[j] = nums[i];
                j++;
            }
        
        return j;
        
    }
}
